src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2529,11 +2524,25 @@ sysroot_finalize_deployment (OstreeSysroot     *self,
         return FALSE;
     }
 
+  const char *osdeploypath = glnx_strjoina ("ostree/deploy/", ostree_deployment_get_osname (deployment));
+  glnx_autofd int os_deploy_dfd = -1;
+  if (!glnx_opendirat (self->sysroot_fd, osdeploypath, TRUE, &os_deploy_dfd, error))
+    return FALSE;
+
+  /* Ensure that the new deployment does not have /etc/.updated or
+   * /var/.updated so that systemd ConditionNeedsUpdate=/etc|/var services run
```

Comment:
Not worth a respin, but just a note: Since `/var` is shared, this applies to all deployments.  It means that if an upgrade is interrupted, on reboot we'll run through `ConditionNeedsUpdate=/var` for the booted deployment too.

In practice, this should be fine; on e.g. Silverblue today that's just `systemd-journal-catalog-update.service`.

Reply:
By a respin do you just mean for the comment? Because obviously unless the sysroot is changed to have /var per-deployment rather than per-os then nothing can be changed there.

My experience is that all the `ConditionNeedsUpdate` units are using it for optimization - they're relatively slow tasks that you don't want to run on every boot. But nothing enforces that and you do need to make the units relatively idempotent if they're performing actions that you really only want to do once (e.g. use a stamp file or something). Just for the record, we make heavy use of `ConditionNeedsUpdate` units at Endless to perform post-boot migration tasks for things that we can't update via the OS update. There's a pile of semi-horrifying things in https://github.com/endlessm/eos-boot-helper using this mechanism.

Reply:
> By a respin do you just mean for the comment? 

Yep.  I would have phrased it like "Ensure the new deployment does not have /etc/.updated, and also delete the stateroot's /var/.updated, so that..." or something.